chunk/chunk_store.go

```diff
@@ -332,25 +338,45 @@ func (c *AWSStore) CreateTables() error {
 	return err
 }
 
-func bigBuckets(from, through model.Time) []int64 {
+func (c *AWSStore) bigBuckets(from, through model.Time) []string {
 	var (
 		secondsInHour = int64(time.Hour / time.Second)
 		fromHour      = from.Unix() / secondsInHour
 		throughHour   = through.Unix() / secondsInHour
-		result        []int64
+
+		secondsInDay = int64(24 * time.Hour / time.Second)
+		fromDay      = from.Unix() / secondsInDay
+		throughDay   = through.Unix() / secondsInDay
+
+		firstDailyBucket = c.dailyBucketsFrom.Unix() / secondsInDay
+		lastHourlyBucket = firstDailyBucket * 24
+
+		result []string
 	)
+
 	for i := fromHour; i <= throughHour; i++ {
-		result = append(result, i)
+		if i > lastHourlyBucket {
+			break
+		}
+		result = append(result, strconv.Itoa(int(i)))
```

Comment:
Not sure the conversion to int is safe here - they'll run out on 2038-01-19 IIRC.  Should probably use https://golang.org/pkg/strconv/#FormatInt